cmd/exec_unix.go

```diff
@@ -0,0 +1,22 @@
+// +build linux darwin
+
+package cmd
+
+import (
+	osexec "os/exec"
+	"syscall"
+)
+
+func exec(command string, args []string, env []string) error {
+	argv0, err := osexec.LookPath(command)
+	if err != nil {
+		return err
+	}
+
+	argv := make([]string, 0, 1+len(args))
+	argv = append(argv, command)
```

Comment:
Alternatively, argv[0] = command, copy(argv[1:], args...)

Reply:
A bit too much magic, I'll keep the sequential form if you don't mind (it just saves one dynamic memory allocation which is blanked out right away when `syscall.Exec` is called.